lib/nrf_modem_lib/nrf_modem_os.c

```diff
@@ -568,12 +567,19 @@ void nrf_modem_os_init(void)
 
 int32_t nrf_modem_os_trace_put(const uint8_t * const data, uint32_t len)
 {
-#ifdef CONFIG_NRF_MODEM_LIB_TRACE_ENABLED
-	int err = nrf_modem_lib_trace_process(data, len);
+	int err;
 
-	if (err != 0) {
-		LOG_ERR("nrf_modem_lib_trace_process returns %d", err);
+#ifdef CONFIG_NRF_MODEM_LIB_TRACE_ENABLED
+	err = nrf_modem_lib_trace_process(data, len);
+	if (err) {
+		LOG_ERR("nrf_modem_lib_trace_process failed, err %d", err);
 	}
 #endif
+
+	err = nrf_modem_trace_processed_callback(data, len);
+	if (err) {
+		LOG_ERR("nrf_modem_trace_processed_callback failed, err %d", err);
+	}
```

Comment:
Documentation for `nrf_modem_trace_processed_callback` states that the function should be invoked again with the same parameters in the event that it returns `-NRF_EAGAIN`. As far as I can see, this is not done here. Am I missing something?

Reply:
It is correct that the `nrf_modem_trace_processed_callback()` can return `-NRF_EAGAIN`. But since the app is processing the trace in interrupt context, there is nothing it can do about this error other than logging it. 

In our testing, we have never seen this error reported by the `nrf_modem_trace_processed_callback()`.

Reply:
As per current design, the `nrf_modem_trace_processed_callback()`  is called in the same context as `trace_put()` function libmodem. So the semaphore which `nrf_modem_trace_processed_callback()` waits for is always available. Hence it will not return `-NRF_EAGAIN`